test/dotcom/trip_plan/location_nudger_test.exs

```diff
@@ -0,0 +1,178 @@
+      # Setup
+      street_address = Faker.Util.pick(@state_house_address_strings)
+      zip_code = Faker.Util.pick(@state_house_zip_codes)
+
+      non_boston_city =
+        fake_with_blocklist(
+          fn -> Faker.Address.city() end,
+          ["Boston"]
+        )
+
+      location = %{
+        name:
+          "#{Faker.Company.name()}, #{street_address}, #{non_boston_city}, MA, #{zip_code}, USA",
+        latitude: Faker.Address.latitude(),
+        longitude: Faker.Address.longitude()
+      }
+
+      # Exercise
+      nudged_location = nudge_location(location)
+
+      # Verify
+      assert nudged_location == location
+    end
+  end
+
+  describe "nudge/1" do
+    test "keeps the 'to' and 'from' fields the same for typical addresses" do
+      # Setup
+      form_data = %{
+        from: %{
+          name: Generators.Address.address(),
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        },
+        to: %{
+          name: Generators.Address.address(),
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        }
+      }
+
+      # Exercise
+      nudged_form_data = nudge(form_data)
+
+      # Verify
+      assert nudged_form_data == form_data
+    end
+
+    test "nudges the 'from' field when needed" do
+      street_address = Faker.Util.pick(@state_house_address_strings)
+      zip_code = Faker.Util.pick(@state_house_zip_codes)
+
+      # Setup
+      form_data = %{
+        from: %{
+          name: "#{Faker.Company.name()}, #{street_address}, Boston, MA, #{zip_code}, USA",
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        },
+        to: %{
+          name: Generators.Address.address(),
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        }
+      }
+
+      # Exercise
+      nudged_form_data = nudge(form_data)
+
+      # Verify
+      assert @state_house_location = nudged_form_data.from
+      assert nudged_form_data.to == form_data.to
+    end
+
+    test "nudges the 'to' field when needed" do
+      street_address = Faker.Util.pick(@state_house_address_strings)
+      zip_code = Faker.Util.pick(@state_house_zip_codes)
+
+      # Setup
+      form_data = %{
+        from: %{
+          name: Generators.Address.address(),
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        },
+        to: %{
+          name: "#{Faker.Company.name()}, #{street_address}, Boston, MA, #{zip_code}, USA",
+          latitude: Faker.Address.latitude(),
+          longitude: Faker.Address.longitude()
+        }
+      }
+
+      # Exercise
+      nudged_form_data = nudge(form_data)
+
+      # Verify
+      assert @state_house_location = nudged_form_data.to
+      assert nudged_form_data.from == form_data.from
+    end
+  end
+
+  defp fake_with_blocklist(fake_fun, blocklist) do
```

Comment:
This feels general-purpose enough that it ought to live somewhere in `Test.Support`. I could make a special-purpose module just for it, or put it with other like things (I couldn't think of any OTOH).

Also open to suggestions for better names for this.

Reply:
I moved it :)